src/coprocessor/codec/batch/lazy_column.rs

```diff
@@ -0,0 +1,552 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: Remove this.
+#![allow(dead_code)]
+
+use std::convert::TryFrom;
+
+use smallvec::SmallVec;
+
+use cop_datatype::{EvalType, FieldTypeAccessor, FieldTypeFlag};
+use tipb::schema::ColumnInfo;
+
+use super::VectorValue;
+use crate::coprocessor::codec::mysql::Tz;
+use crate::coprocessor::codec::{datum, Error, Result};
+
+pub enum LazyBatchColumn {
+    /// Ensure that small datum values (i.e. Int, Real, Time) are stored compactly.
+    /// Notice that there is an extra 1 byte for datum to store the flag, so there are 9 bytes.
+    Raw(Vec<SmallVec<[u8; 9]>>),
```

Comment:
> Will `Raw` store big datum values like json?

Yes. `Raw` can store anything that doesn't need to be decoded during the process. `SmallVec` is just a Vec that stores small values **inline** but also support storing large values in the heap. So it can hold data of any length (just like a normal Vec), but will be highly efficient (compared to Vec) when storing small values.

> What if need returns a JSON column while needn't decode it?

It won't be decoded, as answered above,